src/js/utils/utils.js

```diff
@@ -274,9 +273,44 @@ const utils = {
                     genre       : metadata.genre.map((meta) => utils.stripAccents(meta.toLowerCase()))
                 };
 
-                callback(metadata);
+                if(metadata.duration === 0) {
+
+                    utils.getAudioDuration(track, (err, duration) => {
+
+                        if(err) console.warn(duration);
+
+                        metadata.duration = duration;
+                        callback(metadata);
+                    });
+
+                } else {
+                    callback(metadata);
+                }
             });
         }
+    },
+
+    /**
+     * Get the duration of an audio file with the Audio element
+     *
+     * @param path (string)
+     * @return float
+     */
+    getAudioDuration(path, callback = () => {}) {
+
+        const audio = new Audio;
+
+        audio.addEventListener('loadedmetadata', () => {
+
```

Comment:
> The <audio> element includes a preload attribute. This can be set to "none" or "metadata" which should prevent the audio preloading.

See http://stackoverflow.com/a/36792488/1600598